model/task_lifecycle.go

```diff
@@ -564,55 +530,26 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
 	// does not occur if there's a failed task)
 	if buildComplete {
 		if !failedTask {
-			if pushTaskExists { // this build has a push task associated with it.
-				if pushCompleted && pushSuccess { // the push succeeded, so mark the build as succeeded.
-					err = b.MarkFinished(evergreen.BuildSucceeded, finishTime)
-					if err != nil {
-						err = errors.Wrap(err, "Error marking build as finished")
-						grip.Error(err)
-						return err
-					}
-					updates.BuildNewStatus = evergreen.BuildSucceeded
-
-				} else if pushCompleted && !pushSuccess { // the push failed, mark build failed.
-					err = b.MarkFinished(evergreen.BuildFailed, finishTime)
-					if err != nil {
-						err = errors.Wrap(err, "Error marking build as finished")
-						grip.Error(err)
-						return err
-					}
-					updates.BuildNewStatus = evergreen.BuildFailed
-				}
-
-				// Otherwise, this build does have a "push" task, but it hasn't finished yet
-				// So do nothing, since we don't know the status yet.
+			if err = b.MarkFinished(evergreen.BuildSucceeded, finishTime); err != nil {
+				err = errors.Wrap(err, "Error marking build as finished")
+				grip.Error(err)
+				return err
+			}
+			updates.BuildNewStatus = evergreen.BuildSucceeded
 
-				if err = MarkVersionCompleted(b.Version, finishTime); err != nil {
-					err = errors.Wrap(err, "Error marking version as finished")
-					grip.Error(err)
-					return err
-				}
-			} else { // this build has no push task. so go ahead and mark it success/failure.
-				if err = b.MarkFinished(evergreen.BuildSucceeded, finishTime); err != nil {
-					err = errors.Wrap(err, "Error marking build as finished")
-					grip.Error(err)
-					return err
-				}
-				updates.BuildNewStatus = evergreen.BuildSucceeded
-
-				if evergreen.IsPatchRequester(b.Requester) {
-					if err = TryMarkPatchBuildFinished(b, finishTime, updates); err != nil {
-						err = errors.Wrap(err, "Error marking patch as finished")
-						grip.Error(err)
-						return err
-					}
-				}
-				if err = MarkVersionCompleted(b.Version, finishTime); err != nil {
-					err = errors.Wrap(err, "Error marking version as finished")
+			if evergreen.IsPatchRequester(b.Requester) {
+				if err = TryMarkPatchBuildFinished(b, finishTime, updates); err != nil {
+					err = errors.Wrap(err, "Error marking patch as finished")
 					grip.Error(err)
 					return err
 				}
 			}
+			if err = MarkVersionCompleted(b.Version, finishTime); err != nil {
```

Comment:
This duplicates 568-572, so I think this can be deleted, and that block can be outdented.

Reply:
done